tracer/src/Datadog.Trace/Debugger/Instrumentation/Registry/ProbeMetadataInfo.cs

```diff
@@ -0,0 +1,29 @@
+// <copyright file="ProbeMetadataInfo.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Reflection;
+using Datadog.Trace.Debugger.Expressions;
+using Datadog.Trace.Debugger.Helpers;
+using Datadog.Trace.Debugger.RateLimiting;
+
+namespace Datadog.Trace.Debugger.Instrumentation.Registry
+{
+    /// <summary>
+    /// Holds data needed during Debugger instrumentation execution.
+    /// </summary>
+    internal readonly record struct ProbeMetadataInfo(string ProbeId, AdaptiveSampler Sampler, ProbeProcessor Processor)
```

Comment:
Nit: it's nice that the name here is consistent with `MethodMetadataInfo` because they are used in a similar name, but, this record doesn't really contain "metadata info" per say? Perhaps just `ProbeData` would make more sense, but I don't feel strongly about it nor feel it would be a huge improvement.

Reply:
Done in f2e39f19694d2e9a44c777ae15e0fa38a3d73526.